src/adapters/mqtt/index.ts

```diff
@@ -64,12 +66,12 @@ class MqttAdapter extends Adapter {
       },
       keepalive: serverBinding?.keepAlive,
       username: userAndPasswordSecurityReq
-        ? mqttOptions?.authentication?.userPassword.username
+        ? mqttOptions?.userPassword.username
```

Comment:
Regarding flattening:

```suggestion
        ? mqttOptions?.username
```